### PR TITLE
[EFR32] support softwarediagnostics thread Metrics

### PR DESCRIPTION
**Pull request by jmartinez-silabs**
#### Problem
EFR32 platform doesn't support the attribute threadMetrics

#### Change overview
Add the implementation to encode the attribute base of the available OS tasks stats

#### Testing
Manual test 
./chip-tool softwarediagnostics read thread-metrics 537 0

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

File: src/platform/EFR32/DiagnosticDataProviderImpl.cpp [hunk 1]
```diff
@@ -92,6 +92,60 @@ CHIP_ERROR DiagnosticDataProviderImpl::ResetWatermarks()
...
+        arraySize = uxTaskGetSystemState(taskStatusArray, arraySize, &dummy);
+        /* For each populated position in the taskStatusArray array,
+           format the raw data as human readable ASCII data. */
+
+        for (x = 0; x < arraySize; x++)
+        {
+            ThreadMetrics * thread = new ThreadMetrics();
+
+            strncpy(thread->NameBuf, taskStatusArray[x].pcTaskName, kMaxThreadNameLength - 1);
+            thread->NameBuf[kMaxThreadNameLength] = '\0';
```
Comment (bzbarsky-apple) [inline src/platform/EFR32/DiagnosticDataProviderImpl.cpp:118]: Please just use `Platform::CopyString` for this: `CopyString(thread->NameBuf, taskStatusArray[x].pcTaskName)` I expect will do the right thing.

**Review by bzbarsky-apple — APPROVED** [review]

File: src/platform/EFR32/DiagnosticDataProviderImpl.cpp [hunk 2]
```diff
@@ -92,6 +92,60 @@ CHIP_ERROR DiagnosticDataProviderImpl::ResetWatermarks()
...
+    if (taskStatusArray != NULL)
+    {
+        /* Generate raw status information about each task. */
+        arraySize = uxTaskGetSystemState(taskStatusArray, arraySize, &dummy);
+        /* For each populated position in the taskStatusArray array,
+           format the raw data as human readable ASCII data. */
+
+        for (x = 0; x < arraySize; x++)
+        {
+            ThreadMetrics * thread = new ThreadMetrics();
```
Comment (andy31415) [inline src/platform/EFR32/DiagnosticDataProviderImpl.cpp:115]: Should we check for alloc failure?

**Review by andy31415 — APPROVED** [review]

File: src/platform/EFR32/DiagnosticDataProviderImpl.cpp (hunk 2 above)
Comment (jmartinez-silabs) [reply src/platform/EFR32/DiagnosticDataProviderImpl.cpp:115]: Added the check.

File: src/platform/EFR32/DiagnosticDataProviderImpl.cpp (hunk 1 above)
Comment (jmartinez-silabs) [reply src/platform/EFR32/DiagnosticDataProviderImpl.cpp]: Forgot about copystring. I replaced the two lines by copystring. Thanks.

**Pull request merged**